test/routing_test.rb

```diff
@@ -1452,4 +1452,22 @@ def authorize(username, password)
     end
     get '/users/1/status'
   end
+
+  it 'treats routes with and without trailing slashes differently' do
+    mock_app do
+      get '/foo' do
+        'Foo'
+      end
+
+      get '/foo/' do
+        'Foo with a slash'
+      end
+    end
+
+    get '/foo'
+    assert_equal 'Foo', body
```

Comment:
Should there be a negative test for this too? something like "assert not equal 'Foo with a slash'"?